packages/x-data-grid/src/utils/createSelector.ts

```diff
@@ -1,10 +1,18 @@
 import * as React from 'react';
-import { createSelector as reselectCreateSelector, Selector, SelectorResultArray } from 'reselect';
+import { lruMemoize, createSelectorCreator, Selector, SelectorResultArray } from 'reselect';
 import type { GridCoreApi } from '../models/api/gridCoreApi';
 import { warnOnce } from '../internals/utils/warning';
 
 type CacheKey = { id: number };
 
+const reselectCreateSelector = createSelectorCreator({
+  memoize: lruMemoize,
+  memoizeOptions: {
+    maxSize: 1,
+    equalityCheck: Object.is,
```

Comment:
`NaN` values were previously preventing memoization.
`reselect@5` added a runtime dev check that caught this, though it wouldn't have happened if they had used `Object.is` in the first place (which is the only valid equality comparison operator in JS). Also their dev checks are quite expensive, the grid could feel slower in dev mode.

Not the first time I find questionable decisions in `reselect`.

Reply:
Do you think it makes sense to replace reselect with our own composable selectors implementation?

Reply:
Would be nice but I'd place it as low priority, I don't think it would have a big impact on perf or bundle size.